Modules/Rat_Board.py

```diff
@@ -0,0 +1,197 @@
+
+
+class RescueNotFoundException(RescueBoardException):
+    """
+    Exception raised when a Rescue was not found
+    """
+    pass
+
+
+class RescueNotChangedException(RescueBoardException):
+    """
+    Exception raised when there was no net change to a Rescue.
+    """
+    pass
+
+
+class IndexNotFreeError(RescueBoardException):
+    """ Exception raised when someone attempts to attach a Rescue to a board index in-use"""
+    pass
+
+
+class RatBoard(object):
+    """
+    A rescue board
+    """
+
+    def __init__(self, handler=None):
+        """
+        Create a new Rescue Board.
+
+        Args:
+            handler (): WS API handler
+        """
+        self.handler = handler
+        self._rescues = {}
+        """Rescue objects tracked by this board"""
+
+    def find_by_index(self, index: int) -> Rescue or None:
+        """
+        Searches for and returns a Rescue at a given `index` position, should it exist
+        Args:
+            index (int): case number to return
+
+        Returns:
+            Rescue: rescue found
+            None: no rescue found
+
+        """
+        try:
+            found = self._rescues[index]
+        except KeyError:
+            # the key doesn't exist, therefore no rescue at that index.
+            pass
+        else:
+            # we found something
+            return found
+
+    def find_by_name(self, client: str) -> Rescue or None:
+        """
+        Searches for and returns a Rescue for a given client, should it exist.
+
+        Returns:
+            Rescue: found rescue
+            None:   no such rescue
+        """
+        for index, rescue in self._rescues.items():
+            if rescue.client == client:
+                return rescue
+
+    def find_by_uuid(self, guid: UUID) -> Rescue or None:
+        """
+        Searches for and returns a rescue by api ID, should it exist.
+
+        Args:
+            guid (UUID): uuid to search for
+
+        Returns:
+            Rescue: found rescue
+            None:   no rescue found
+
+        """
+        for index, rescue in self._rescues.items():
+            if rescue.case_id == guid:
+                return rescue
+
+    def search(self, index: int = None, client: str = None, api_id: UUID = None) -> Rescue or None:
+        """
+        Searches for and returns a rescue for a given Index / client name / api id, should it exist
+            Will match the first found case matching at least one of the parameters.
+
+        At least one of the parameters must not be None.
+
+        Args:
+            index (int): Rescue at given board index, may be None
+            client (str): client's commander name / irc nickname, may be None
+            api_id (UUID): api id of rescue, may be None
+
+        Returns:
+            Rescue: Found rescue on the board, otherwise None
+
+        Raises:
+            ValueError: command invoked without params, or all params are None
+        """
+        raise NotImplementedError
```

Comment:
Best to kill the method entirely. Afaik nothing's using it yet and raising NotImplementedError instead doesn't really help.

Reply:
Its in the process of being reimplemented to accept a string. the NotImplementedError is temporary >.>